google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java

```diff
@@ -1279,6 +1281,40 @@ public Boolean call() {
     }
   }
 
+  @Override
+  public Map<BucketInfo.LifecycleRule, Boolean> deleteLifecycleRules(
+      String bucket, BucketInfo.LifecycleRule... rules) {
+    final Storage storage = getOptions().getService();
+    final Map<BucketInfo.LifecycleRule, Boolean> results = Maps.newHashMap();
+    try {
+      Bucket remoteBucket =
+          storage.get(bucket, Storage.BucketGetOption.fields(Storage.BucketField.LIFECYCLE));
+      ArrayList<BucketInfo.LifecycleRule> lifecycleRules =
+          new ArrayList(remoteBucket.getLifecycleRules());
+      Iterator<BucketInfo.LifecycleRule> ruleIterator = lifecycleRules.iterator();
+      for (BucketInfo.LifecycleRule rule : rules) {
+        while (ruleIterator.hasNext()) {
+          BucketInfo.LifecycleRule actualRule = ruleIterator.next();
+          if (rule.getAction().getActionType().equals(actualRule.getAction().getActionType())) {
+            ruleIterator.remove();
+            results.put(rule, Boolean.TRUE);
+          }
+        }
+      }
```

Comment:
This will not always work:
```
Iterator it = ...
for (rule: rules) {
   while( it.hasNext() ) {
     // once it reaches the finish it will never start again
   }
}
```
More safety would be:


```
List<Rules> newRules = new ArrayList<>();
for (Rule r: remoteBucket.getLifecycleRules()) {
  for(Rule d: rulesToDelete) {
     if ( delete ) {
        ...
     } else {
       newRules.add(r);
     }
  }
}

```

Reply:
done with different approach to fix the `concurrentmodificationexception`.